File: solutions/0463. Island Perimeter/0463.py
# Time complexity: O(mn)
# Space complexity: O(1)
class Solution:
  def islandPerimeter(self,grid: List[List[int]]) -> int:
    ans = 0
    for i in range(len(grid)):
      for j in range(len(grid[0])):
        if grid[i][j] == 1:
          ans += (i == 0 or grid[i - 1][j] == 0)
          ans += (j == 0 or grid[i] [j - 1] == 0)

    return 2 * ans      

File: solutions/0463. Island Perimeter/0463.cpp
// Time complexity: O(mn)
// Space complexity: O(1)
class Solution {
 public:
  int islandPerimeter(const vector<vector<int>>& grid) {
    int ans = 0;
    for (int i = 0; i < grid.size(); ++i)
      for (int j = 0; j < grid[0].size(); ++j)
        if (grid[i][j] == 1) { 
          ans += (i == 0 || grid[i - 1][j] == 0);
          ans += (j == 0 || grid[i][j - 1] == 0);
        }

    return 2 * ans;
  }
};